### PR TITLE
fix(lucky drop): avoid compound inject dialog

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-1645
https://mask.atlassian.net/browse/MF-1694

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 1]
```diff
@@ -465,39 +467,152 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
...
+                                    </div>
+                                    <Typography className={classNames(classes.selectAllCheckBoxText)}>
+                                        {tr('select_all')}
+                                    </Typography>
+                                </div>
                                 <Typography>
-                                    {loadingToken ? t.loading_token() : searched ? t.search_no_result() : null}
+                                    <Translate.nft_shift_select_tip
+                                        components={{
+                                            text: <span style={{ color: '#1C68F3' }} />,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: Can we read it from `theme.palette.maskColor`?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx [hunk 2]
```diff
@@ -290,131 +299,133 @@ export function RedPacketERC721Form(props: RedPacketERC721FormProps) {
...
                     contract={contract}
-                    open={openConfirmDialog}
                     tokenList={tokenDetailedList}
-                    onBack={() => setOpenConfirmDialog(false)}
+                    onBack={() => setOpenNFTConfirmDialog(false)}
                     onClose={onClose}
                 />
-            ) : null}
+            ) : (
+                <>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx]: Does it better to have a sub-component for this?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 3]
```diff
@@ -429,15 +424,22 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
     }, [searchTokenListInput])
 
     useEffect(() => {
+        const tokenIdOwnerList = tokenDetailedOwnerList.map((t) => t.tokenId)
         setNonExistedTokenIdList(
-            tokenDetailedSelectedList.filter((token) => !isSameAddress(token.ownerId, account)).map((t) => t.tokenId),
+            tokenIdFilterList
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: Does it necessary to de-dup token ids?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 4]
```diff
@@ -465,39 +467,152 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
...
+                                    : t.nft_max_shares(maxSharesOptions)}
+                            </Typography>
+                        }
+                        placement="top-end"
+                        classes={{ tooltip: classes.tooltip, arrow: classes.arrow }}
+                        arrow>
+                        <QuestionMarkIcon className={classes.questionMarkIcon} />
+                    </ShadowRootTooltip>
+                    <Typography>
+                        <span className={classes.selectedTokenAmount}>{tokenDetailedSelectedList.length}</span> NFTs
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: i18n

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 5]
```diff
@@ -465,39 +467,152 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
...
+            </Button>
+        </DialogContent>
+    ) : (
+        <DialogContent className={classNames(classes.dialogContent, classes.dialogContentFixedHeight)}>
+            <div className={classes.searchWrapper}>
+                <Paper className={classes.search} elevation={0}>
+                    <Icons.Search className={classes.iconButton} />
+                    <InputBase
+                        value={tokenDetailedOwnerList.length === 0 ? searchTokenListInput : tokenIdListInput}
+                        placeholder="Token ID separated by comma, e.g. 1224, 7873, 8948"
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: i18n

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx [hunk 6]
```diff
@@ -237,115 +235,110 @@ export function RedpacketNftConfirmDialog(props: RedpacketNftConfirmDialogProps)
...
+                            <ActionButton
+                                size="medium"
+                                loading={isSending}
+                                disabled={isSending}
+                                onClick={onSendTx}
+                                className={classNames(classes.button, classes.sendButton)}
+                                fullWidth>
+                                {t.send_symbol({
+                                    amount: tokenList.length.toString(),
+                                    symbol: tokenList.length > 1 ? 'NFTs' : 'NFT',
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx]: Have a try:

https://www.i18next.com/translation-function/plurals#singular-plural

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx (hunk 3 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: o yes

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx (hunk 2 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx]: Refactored as `if (...) return <Component />` way.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**